docs/source/VUDRM-key-provision.md

```diff
@@ -0,0 +1,442 @@
+
+## Key provider API
+
+The key provider API will provide all the information required in order to encrypt various types of content. In order to be as flexible as possible it will return encryption keys in Base16 and Base64 as some systems require different values. 
+
+The following sections describe the requests and responses in order to retrieve the keys. The most common scenario is to request CENC and Fairplay encryption keys. 
+
+### Request
+
+A `GET` should be made to the following URL:
+
+```URL
+https://key-provider.drm.technology/<DRM_TYPE>/<CLIENT_NAME>/<CONTENT_ID> 
+```
+
+The breakdown of this URL is:
+- `DRM_TYPE`: The type of encryption keys being requested. Possible values are `cenc`, `fairplay`, `widevine`, `playready`, and `aes`. See [Using the encryption keys](#using-the-encryption-keys) for more information.
+- `CLIENT_NAME`: The account name. Plese contact support@vualto.com if you do not have an account name.
+- `CONTENT_ID`: A unique content identifier. This value will always generate the same encryption keys.
+
+In order to provide the keys securely, an API key is required in the header of the request. Please contact support@vualto.com if you do not have an API key.
+
+Below is an example curl request for `cenc` encryption keys using the client `vualto` and with the `CONTENT_ID` of `test`.
+
+```bash
+curl -X "GET" "https://key-provider.drm.technology/cenc/vualto/test" -H "API_KEY: <API_KEY>" 
+```
+
+### Response
+
+In order to provide the keys securely, the DRM encryption keys are encrypted in the response.
+
+The format of the response will be:
+
+```JSON
+{
+    "key":"r8tXpf8wSSHKVnW1fz+MgZY3BTnTO+/IGFglt9oUwtKWj8eyguSLd0bzOhcn4DMF4JWOAbhbKopJE/cZWPqIfl3RCIwl46UP57/m7870+z3NWxh/JSvrfWBq4SGmkykfDLKjyLBqb5F6dDlUB+flcfZMcQh6FGk5GNGEniXliB/kyCrVDiKdwAw3hft8rT4/itFQDMRKkhJf1Fh67AZ1LyzOI3CCY/oO4w/XF/XMAJ1z92tAKULI+cPMFaXT3I77N3iaQoYN78mJkKgAr71Tlf91+ASB8jqOCHD6nNgm6nGxZlsTVK5wxdhT4zbMJq4xb7daSy7xMWdH/1eXuIh0ZhWicKJqbWHKIeifZWFras/0QzqN27rupHF3UYnYQ40V3ESE2PUIe8Cs8471QRHlruYlwtgBBmGme2ERZWFjrRFEeUt8SobQkCIZrzDEKaQ2bJMyOy1yMt8oklpFaW9pBg3yHZEK1WZn5u8ynV+ZWLI6soCNpMgkPYe2UtnFOhFRZfT8WG09FGJiouzyL3fCZAguxP8u9jkzQTv15UhwO/StdpTKCn1yxr3KoyxGAk3L|166c43d4023880a99691fd9679e6ad785305f205"
+} 
+```
+
+The value of `key` has two components separated by a pipe (ASCII code 124):
+- Encrypted blob containing the DRM encryption keys. 
+- Hash used in a checksum.
+
+#### Decrypting the response
+
+This section contains examples of how to decrypt the response from the key provider API.
+
+You will need the `CLIENT_NAME`, `SHARED_SECRET` and the `KEY_PROVIDER_RESPONSE` value from the request to the Keyprovider API.
+
+##### C#
+
+```c#
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Vudrm.EncryptionExamples
```

Comment:
Why is it called Vudrm.EncryptionExamples when all we are showing is decryption :)

Reply:
Where the code examples come from contains the methods for encryption and decryption.